File: python/2798.py
n, m = map(int, input().split())

a = list(map(int, input().split()))
b = []

for i in range(0, n-2):
    for j in range(i+1, n-1):
        for k in range(j+1, n):
            result = a[i]+a[j]+a[k]
            if(result <= m):
                b.append(result)

print(max(b))

File: js/2798.js
const fs = require('fs');

const input = fs.readFileSync('/dev/stdin').toString().split('\n');

const [n, m] = input[0].split(' ').map(Number);
const arr = input[1].split(' ').map(Number);

let result = 0;

for (let i = 0; i < n; i++) {
  for (let j = 0; j < n; j++) {
    for (let k = 0; k < n; k++) {
      if (!(i === j || j === k || i === k)) {
        const sum = arr[i] + arr[j] + arr[k];
        if (sum <= m && result < sum) {
          result = sum;
        }
      }
    }
  }
}

console.log(result);
